File: CS120/assignments/02/java/pi.java
import java.lang.Math;

class PiApproximation
{
	private static double radius = 1;

	private static double height(double base, double hypotenuse)
	{
		return(Math.sqrt(hypotenuse*hypotenuse - base*base));
	}

	private static double circle_pi(int rectangles)
	{
		double sum, width;

		sum = 0;
		width = radius/rectangles;

		for (int i = 0; i < rectangles; i++)
		{
			sum += width*height((0.5+i)*width, radius);
		}

		return(4*sum);
	}

	private static double leibniz_pi(int iterations)
	{
		int denominator = 1;
		double sum = 0;

		while (denominator < iterations*2)
		{
			if ((denominator/2) % 2 == 0)
			{
				sum += 1.0/denominator;
			}
			else
			{
				sum -= 1.0/denominator;
			}

			denominator += 2;
		}

		return(4*sum);
	}

	public static void main(String[] args)
	{
		System.out.println("Approximations for pi");
		System.out.println("Iterations      Circle Method   Leibniz Method");
		System.out.println("----------------------------------------------");

		for (int i = 1; i <= 1000000; i *= 10)
		{
			double circle_result = circle_pi(i);
			double leibniz_result = leibniz_pi(i);
			System.out.println(String.format("%10d%20.12f%16.12f", i, circle_result, leibniz_result));
		}
	}
}

File: cs120/assignments/02/python/pi.py
#!/usr/bin/env python

import math

# Lazy and using this as a global. This might not actually be a good idea.
radius = 1.0

def height(base, hypotenuse):
	return(math.sqrt(hypotenuse**2 - base**2))

def circle_pi(rectangles):
	sum = 0.0
	width = radius/rectangles
	for i in range(rectangles):
		sum += width * height(((0.5+i) * width), radius)

	return(4*sum)

def leibniz_pi(iterations):
	sum = 0
	denominator = 1
	for i in range(iterations):
		if ((denominator//2) %2) == 0:
			sum += 1.0/denominator
		else:
			sum += -1.0/denominator
		denominator += 2

	return(4*sum)

# Function definitions done, now for the program itself.

print("Approximations for pi")
print("Iterations      Circle Method   Leibniz Method")
print("----------------------------------------------")

i = 1
while i <= 1000000:
	print("%(index)10d%(circle_result)20.12f%(leibniz_result)16.12f" % \
		{"index": i, \
		"circle_result": circle_pi(i), \
		"leibniz_result": leibniz_pi(i)})
	i*= 10


